packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -292,9 +292,15 @@ export function attributeToPropertyName(element: IRElement, attrName: string): s
     let propName = attrName;
     if (shouldCamelCaseAttribute(element, attrName)) {
         const attrToSplit = ATTRS_PROPS_TRANFORMS[propName] || propName;
-        propName = attrToSplit
+
+        // flag indicating whether leading character should be uppercased
+        const pascalCase = attrToSplit.startsWith('-');
+
+        // remove leading hyphen
+        const normalizedAttrToSplit = pascalCase ? attrToSplit.replace(/^[-]+/, '') : attrToSplit;
+        propName = normalizedAttrToSplit
             .split('_')
-            .map(camelcase)
+            .map(input => camelcase(input, { pascalCase }))
```

Comment:
Because the casing rule is getting more and more complex, I would recommend ditching the `camelcase` package and rollout our own attribute to property mapping. It would give us finer control here and would avoid having to split the attribute name and join on `_`.

```js
function attrToProp(attr) {
	let prop = '';
  let shouldUpperCaseNext = false;
  
  for (let i = 0; i < attr.length; i++) {
  	const char = attr.charAt(i);
    
    if (char === '-') {
    	shouldUpperCaseNext = true;
    } else {
    	prop += shouldUpperCaseNext ? char.toUpperCase() : char;
      shouldUpperCaseNext = false;
    }
  }
  
  return prop;
}
```

Link: [Fiddle with test](https://jsfiddle.net/pmdartus/uh1njmdx/7/)